assets/src/dashboard/app/font/fontProvider.js

```diff
@@ -57,15 +49,26 @@ function FontProvider({ children }) {
 
   useEffect(() => {
     let mounted = true;
-    getAllFonts().then((newFonts) => {
-      if (mounted) {
-        setFonts(newFonts);
-      }
-    });
+    async function loadFonts() {
+      const newFonts = await import(
+        /* webpackChunkName: "chunk-fonts" */ '../../../fonts/fonts'
+      );
+      const formattedFonts = newFonts.default.map((font) => ({
+        name: font.family,
+        value: font.family,
+        ...font,
+      }));
+
+      setFonts(formattedFonts);
+    }
+    if (mounted) {
+      loadFonts();
+    }
```

Comment:
This doesn't really make sense - JS is serial and the value of `mounted` cannot change between the initialization just above and this point. `loadFonts` can be unwrapped and just put straight into the `useEffect` here.

Reply:
The reason it is wrapped it because of the async behaviour.

Reply:
My point here was, that in a setup as this:

```js
let mounted = true;
if (mounted) {

}
return () => { mounted = false; };
```

The variable will always still be true in the if-statement when execution reaches that point, as this will run synchronously and the unload function cannot run in-between as it's literally not defined yet.

Reply:
@barklund How would you write this statement then?